### PR TITLE
[BEAM-5649] Conditionally remove CREATE_VIEW from java generated protos.

### PR DESCRIPTION
**Pull request by robertwb**
The graph generation is tightly coupled to legacy runner execution, but this
makes it possible for these implementation details to not leak outside of the
Java SDK.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by robertwb** [comment]
Run Java PostCommit

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PipelineTranslation.java [hunk 1]
```diff
@@ -81,8 +96,83 @@ public void visitPrimitiveTransform(Node node) {
...
+      res = elideDeprecatedViews(res);
+    }
     // Validate that translation didn't produce an invalid pipeline.
     PipelineValidator.validate(res);
     return res;
   }
+
+  private static RunnerApi.Pipeline elideDeprecatedViews(RunnerApi.Pipeline pipeline) {
+    // Record data on CreateView operations.
+    Collection<String> viewTransforms = new HashSet<>();
```
Comment (mxm) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PipelineTranslation.java]: Why is this `Collection`? IMHO, `Set` would make the set attributes of the implementation clearer.

Comment (robertwb) [reply runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PipelineTranslation.java]: No good reason. Updating now. (I'm going to ensure the postcommits pass before I push.)

**Comment by mxm** [comment]
@tweise Already done with this PR :)

**Pull request merged**